ClassLibrary/ClassLibrary/Library.cs

```diff
@@ -43,24 +43,24 @@ public void ReaderBoroweBook(int IDReader, String NameBook)
 
            String ISBN = listAllBookWithName[0].ISBN;
 
-            HashSet<Book> allBookWithISBN= new HashSet<Book>(listAllBookWithName);
-
-           HashSet<Book> borrowedBookWithISBN = departmentReaders.BorrowedBooksWithISBN(ISBN);
+           List<BorrowedBook> borrowedBookWithISBN = departmentReaders.BorrowedBooksWithISBN(ISBN);
 
             //Free Book
-            allBookWithISBN.ExceptWith(borrowedBookWithISBN);
+            List<Book> freebooks = listAllBookWithName.FreeBook(borrowedBookWithISBN);
 
-            if (allBookWithISBN.Count==0)
+            if (freebooks.Count==0)
             {
-                throw new ArgumentException("ALL books borrowed");
+                TimeSpan days=departmentReaders.GetTimeWhenFreeBook(borrowedBookWithISBN);
+                throw new ArgumentException($"ALL books borrowed. {days.Days} days when the nearest book is free ");
```

Comment:
"all books borrowed" is not an exceptional situation according to the requirements, you shouldn't throw an exception here.

Reply:
corrected in new version

Reply:
there is a check in the code, but I cannot cover it with a test

Reply:
you can cover anything with tests :)

Reply:
BookAvaible30Days